superset/connectors/sqla/models.py

```diff
@@ -1420,7 +1441,6 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                 and db_engine_spec.allows_hidden_cc_in_orderby
                 and col.name in [select_col.name for select_col in select_exprs]
             ):
-                validate_adhoc_subquery(str(col.expression))
```

Comment:
Moved this [up](https://github.com/apache/superset/pull/19454/files#diff-af5e15a55ff9c81f441eb6b9b10dae13b2ee17fec614d3e3ad0057ed0dc814aeR1155-R1160), where it's still a string.